src/soc/amd/common/boot/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+#![feature(llvm_asm)]
+#![feature(lang_items, start)]
+#![no_std]
+#![no_main]
+#![feature(global_asm)]
+
+use arch::bzimage::BzImage;
+use arch::ioport::IOPort;
+use model::Driver;
+use uart::i8250::I8250;
+mod acpi;
+use acpi::setup_acpi_tables;
+extern crate heapless; // v0.4.x
+use heapless::consts::*;
+use heapless::Vec;
+
+use core::ptr;
+// Until we are done hacking on this, use our private copy.
+// Plan to copy it back later.
+global_asm!(include_str!("bootblock.S"));
+
+fn poke32(a: u32, v: u32) -> () {
```

Comment:
can we move these to arch?

Reply:
I don't like the API much--we definitely shouldn't be using it in general. I only kept them because of the one call in boot()--and even that I want to remove in favor of VolatileCell later.

See also https://www.ralfj.de/blog/2020/12/14/provenance.html for why.